### PR TITLE
Create methods to write and read on files

### PR DESCRIPTION
**Pull request by rica213**
In this milestone, here are what we have done:
- Create a method to write on file
- Create a method to read on file
- Use those previous methods to preserve Books, Person, and Rentals object
- Update readme file

### CODE REVIEW

File: classes/app.rb [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
 
   # option 1 - List all books
   def list_books
     puts 'No book yet!' if @books.empty?
-    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+    @books.each { |book| puts "Title: \"#{book['title']}\", Author: \"#{book['author']}\" Rentals: #{book['rentals']}" }
   end
 
   # option 2 - List all people
   def list_people
```
Comment (julie-ify) [inline classes/app.rb:20]: - [ ] After creating a student or a teacher when I try to list people using option 2, the app crashes. It's only when I restart the app that I am able to list people. Kindly fix that.

  image

File: classes/app.rb [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
+    @books = read_file('./data/books.json')
+    @rentals = read_file('./data/rentals.json')
   end
 
   # option 1 - List all books
   def list_books
     puts 'No book yet!' if @books.empty?
-    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+    @books.each { |book| puts "Title: \"#{book['title']}\", Author: \"#{book['author']}\" Rentals: #{book['rentals']}" }
   end
```
Comment (julie-ify) [inline classes/app.rb:17]: - [ ] After adding a book when I select the option to list books the app crashes. Kindly fix that as well.

  image

File: classes/app.rb [hunk 3]
```diff
@@ -89,22 +91,31 @@ def create_rental
...
     print 'ID of person: '
     id = gets.chomp.to_i
     puts 'Rentals:'
     @rentals.each do |rental|
-      puts "Date: #{rental.date}, Book: \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
+      if rental['person']['id'] == id
+        puts "Date: #{rental['date']}, Book: \"#{rental['book']['title']}\" by #{rental['book']['author']}"
+      end
     end
   end
```
Comment (julie-ify) [inline classes/app.rb:120]: - [ ] After creating a rental and I select option to list the rentals the app crashes. Kindly fix that.

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @rica213 and @shaaibu7,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @rica213 and @shaaibu7 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights

✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rica213** [comment]
@danmainah , @julie-ify ,

Thank you for taking the time in reviewing our project. We are grateful for your hard work, and your insights on how to improve our code. 🙇🏼‍♀️

**Pull request merged**